pam/internal/adapter/newpasswordmodel.go

```diff
@@ -64,6 +62,14 @@ func (m newPasswordModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 		m.Clear()
 		return m, nil
 
+	case newPasswordCheckResult:
+		if msg.msg != "" {
+			m.Clear()
+			return m, sendEvent(errMsgToDisplay{msg: msg.msg})
+		}
+
+		return m, tea.Batch(sendEvent(errMsgToDisplay{}), m.focusNextField())
+
```

Comment:
Like I've said before: we had no decide on this. Why did you change it anyway? It makes no sense to me.

Reply:
Because as I said, it's something that we need to do in three models, native one needs it gdm too and I didn't want to keep track of the current challenge in all of them since we've a central authentication place.

Doing it in each model is a risk of having different behaviors IMHO, and this works well enough once it's been abstracted.